ethereum/eip712/encoding.go

```diff
@@ -0,0 +1,214 @@
+package eip712
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/simapp/params"
+	"github.com/cosmos/cosmos-sdk/x/auth/migrations/legacytx"
+
+	cosmosTypes "github.com/cosmos/cosmos-sdk/types"
+	txTypes "github.com/cosmos/cosmos-sdk/types/tx"
+
+	apitypes "github.com/ethereum/go-ethereum/signer/core/apitypes"
+	ethermint "github.com/evmos/ethermint/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+)
+
+var (
+	ethermintProtoCodec codec.ProtoCodecMarshaler
+	ethermintAminoCodec *codec.LegacyAmino
+)
+
+// The process of unmarshaling SignDoc bytes into a SignDoc object requires having a codec
+// populated with all relevant message types. As a result, we must call this method on app
+// initialization with the app's encoding config.
+func SetEncodingConfig(cfg params.EncodingConfig) {
+	ethermintAminoCodec = cfg.Amino
+	ethermintProtoCodec = codec.NewProtoCodec(cfg.InterfaceRegistry)
+}
+
+// Get the EIP-712 object hash for the given SignDoc bytes by first decoding the bytes into
+// an EIP-712 object, then hashing the EIP-712 object to create the bytes to be signed.
+// See https://eips.ethereum.org/EIPS/eip-712 for more.
+func GetEIP712HashForMsg(signDocBytes []byte) ([]byte, error) {
```

Comment:
go-ethereum now has this as a util so we can remove this function

Reply:
It looks like this util was released in [v1.10.21](https://github.com/ethereum/go-ethereum/releases/tag/v1.10.21) so we cannot access it at the moment